eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Gov.hs

```diff
@@ -240,23 +261,42 @@ actionWellFormed ga = failureUnless isWellFormed $ MalformedProposal ga
       ParameterChange _ ppd -> ppuWellFormed ppd
       _ -> True
 
+-- | Called from within `addAction` to check if the proposal being added is the first one
+-- after a set of contiguous epochs (we need to check only the previous epoch to be sure)
+-- that had no active governance proposals to vote on, so that `DRep` expiry can be updated
+-- and the dormant-epochs counter can be reset to zero.
+-- Although this function is called on each proposal, it makes sure to update the `DRep`
+-- expiry only the first time that a new proposal is accepted in an epoch.
+updateDRepExpiry :: GovRuleState era -> GovRuleState era
+updateDRepExpiry grs@(GovRuleState govSnapshots vState) =
+  let wasPrevEpochDormant = Seq.null . snapshotIds $ govSnapshots ^. prevGovSnapshotsL
+      isCurEpochDormantSoFar = Seq.null . snapshotIds $ govSnapshots ^. curGovSnapshotsL
+      numDormantEpochs = vState ^. vsNumDormantEpochsL
+      isNumDormantEpochsNonZero = numDormantEpochs /= 0
+   in if isNumDormantEpochsNonZero && isCurEpochDormantSoFar && wasPrevEpochDormant
```

Comment:
Please correct me if I'm wrong, but my thought process behind this was: this is triggered _just before_ the processing of the first proposal, and not _after_, and so, the current epoch is still expected to be empty. It will become non-empty as soon as any proposal is added and continue to be non-empty, so if we check that it is non-empty, it will mean that it is applicable for all proposals except the first one. `isCurEpochDormantSoFar` will be `True` when it is empty, and we receive the first proposal action to add to the snapshot. It will be `False` for every proposal action that is submitted after the first one in the current epoch. We only want to act upon the first submission in an epoch, and so we check its truthiness.

Reply:
@teodanciu and @lehins suggested significant improvements in this implementation and the result has been pushed. This conversation can be resolved, as it concerns a stale part of the implementation that has been removed completely.